src/Microsoft.ML.Transforms/Text/NgramHashingTransformer.cs

```diff
@@ -1,33 +1,38 @@
-// Licensed to the .NET Foundation under one or more agreements.
+﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using Float = System.Single;
-
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Reflection;
-using System.Text;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
 using Microsoft.ML.Runtime;
 using Microsoft.ML.Runtime.CommandLine;
 using Microsoft.ML.Runtime.Data;
 using Microsoft.ML.Runtime.Internal.Utilities;
 using Microsoft.ML.Runtime.Model;
 using Microsoft.ML.Transforms.Text;
-using Microsoft.ML.Data;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Text;
 
-[assembly: LoadableClass(typeof(NgramHashingTransformer), typeof(NgramHashingTransformer.Arguments), typeof(SignatureDataTransform),
+[assembly: LoadableClass(NgramHashingTransformer.Summary, typeof(IDataTransform), typeof(NgramHashingTransformer), typeof(NgramHashingTransformer.Arguments), typeof(SignatureDataTransform),
     "Ngram Hash Transform", "NgramHashTransform", "NgramHash")]
 
-[assembly: LoadableClass(typeof(NgramHashingTransformer), null, typeof(SignatureLoadDataTransform),
+[assembly: LoadableClass(NgramHashingTransformer.Summary, typeof(IDataTransform), typeof(NgramHashingTransformer), null, typeof(SignatureLoadDataTransform),
+    "Ngram Hash Transform", NgramHashingTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(NgramHashingTransformer.Summary, typeof(NgramHashingTransformer), null, typeof(SignatureLoadModel),
+    "Ngram Hash Transform", NgramHashingTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(NgramHashingTransformer), null, typeof(SignatureLoadRowMapper),
     "Ngram Hash Transform", NgramHashingTransformer.LoaderSignature)]
 
 namespace Microsoft.ML.Transforms.Text
```

Comment:
Should this stay in Microsoft.ML.Transforms.Text? Or should this move up one level to Microsoft.ML.Transforms?

Reply:
I prefer to keep it as is. All text transforms live in this namespace.

---
In reply to: [238483123](https://github.com/dotnet/machinelearning/pull/1811#discussion_r238483123) [](ancestors = 238483123)